src/swipes.py

```diff
@@ -0,0 +1,342 @@
+    # sum swipes and counters for rows with the same weekday and location
+    final_grouping = counter_rename.groupby(['weekday', 'location', 'session_type']).sum().reset_index()
+    final_grouping['average'] = np.around(np.divide(final_grouping['swipes'], final_grouping['counter']), decimals=2)
+    final_grouping.sort_values(by=['location', 'weekday'])
+    final_grouping.to_csv('{}{}'.format(EATERY_DATA_PATH, output_file), header=True, index=False)
+  except Exception as e:
+    print('Faled at sort_by_day')
+    print('Data update failed:', e)
+
+
+def export_data(file_path):
+  """Transforms our tabular data into custom objects to be placed in Eatery objects
+
+  Keyword arguments:
+  file_path -- the file path to our input csv file with all wait time data, string
+  """
+  global breaks
+  global weekdays
+
+  try:
+    df = pd.read_csv(file_path)
+    data = {}
+    today = date.today()
+    session_type = sort_session_type(today, breaks)
+    for location in df['location'].unique():
+      eatery_name = LOCATION_NAMES[location]['name']
+      # look at information that pertains to today's criteria
+      today_df = df.loc[(df['location'] == location) & (df['weekday'] == weekdays[today.weekday()])]
+
+      if today_df.empty:
+        print('{} has no swipe data for {}'.format(eatery_name, weekdays[today.weekday()]))
+        continue
+
+      # df contains current session
+      if today_df['session_type'].str.contains(session_type, regex=False).any():
+        today_df = today_df.loc[(today_df['session_type'] == session_type)]
+      else:  # not all breaks will be in data to start, default to regular for time being
+        print('defaulting to regular session_type')
+        today_df = today_df.loc[(today_df['session_type'] == 'regular')]
+
+      max_swipes = today_df['average'].max()
+      json_data = json.loads(today_df.to_json(orient='table'))
+
+      for row in json_data['data']:
+        new_timeblock = SwipeDataType(  # represents the data for a single timeblock of swipe data
+            end_time=row['end_time'],
+            session_type=session_type,
+            start_time=row['start_time'],
+            swipe_density=round(row['average'] / max_swipes, SWIPE_DENSITY_ROUND),
+            wait_time_high=row['wait_time_high'],
+            wait_time_low=row['wait_time_low'],
+        )
+
+        if eatery_name not in data:
+          data[eatery_name] = [new_timeblock]
+        else:
+          data[eatery_name].append(new_timeblock)
+
+    return data
+
+  except Exception as e:
+    print('Failed at export_data')
+    print('Data update failed:', e)
+    return {}
+
+def aggregate_breaks(base_df):
+  """Returns the dataframe with wait time calculations done on the combination of all break data.
+
+  Keyword arguments:
+  base_df -- the input dataframe which only contains one location and one day of the week
+
+  NOTE: This method is currently not working as intended so it is not present in the live code.
```

Comment:
lol

Reply:
don't roast me, just wanna get this out my dude